src/imap.rs

```diff
@@ -845,6 +868,18 @@ impl Imap {
         }
     }
 
+    fn message_is_autocrypt_setup(&self, msg: &imap::types::Fetch) -> bool
+    {
+
+        let res =
+        match msg.bodystructure()
```

Comment:
I like the `match`, but alternatively, you could write:

```rust
msg.bodystructure().
    and_then(|bs| body_structure_is_autocrypt_setup(bs)).
    unwrap_or(false)
```

Whatever you find more readable.